lib/mutant/ast/regexp/transformer.rb

```diff
@@ -0,0 +1,82 @@
+module Mutant
+  module AST
+    module Regexp
+      class Transformer
+        include AbstractType
+
+        REGISTRY = {} # rubocop:disable MutableConstant
```

Comment:
> Assuming there's not a compelling reason I would recommend collapsing the implementation and exposing only the Registry public methods in the interface.

By this you mean reducing the registry down to just registration and then replacing instances of `REGISTRY.call` with `REGISTRY.lookup(...).call(...)`?